### PR TITLE
feat : Progress Circle

### PR DESCRIPTION
**Pull request by ahmed-osman3**
https://jira.zebra.com/browse/UX-877

### CODE REVIEW

File: lib/src/components/progress/progress_circle.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import 'dart:math' as math;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import '../../../zeta_flutter.dart';
+import 'progress.dart';
+
+/// Sizes for [ZetaProgressCircle]
+enum ZetaCircleSizes {
```
Comment (mikecoomber) [inline lib/src/components/progress/progress_circle.dart:10]: Is there any way we can use a more generic enum for this like ZetaWidgetSize?

Comment (ahmed-osman3) [reply lib/src/components/progress/progress_circle.dart:10]: Circle sizes have more options so i could use an extension on ZetaWidgetSize , but it feels unnecessary to only extend 3 sizes.

File: lib/src/components/progress/progress_circle.dart [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      ..add(DoubleProperty('progress', progress))
+      ..add(DiagnosticsProperty<bool>('rounded', rounded));
+  }
+}
+
+///Class definition for [ZetaProgressCircleState]
+class ZetaProgressCircleState extends ZetaProgressState<ZetaProgressCircle> {
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
```
Comment (thelukewalton) [inline lib/src/components/progress/progress_circle.dart:59]: We need to wrap this in a fixed size box, as there are no constraints. This is causing Widgetbook to fail, and would be a paint using the component in an app.
Wrap AnimatedBuilder in constraints like this: 
```dart
 return ConstrainedBox(
      constraints: BoxConstraints.tight(_getSize()),
      child: AnimatedBuilder(
```

File: lib/src/components/progress/progress_circle.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  CirclePainter({this.progress = 0, this.rounded = true});
+
+  ///Percentage of progress in decimal value, defaults to 0
+  final double progress;
+
+  ///Is circle rounded, defaults to true
+  final bool rounded;
+
+  final _paint = Paint()
+    ..color = Colors.blue
```
Comment (thelukewalton) [inline lib/src/components/progress/progress_circle.dart:110]: here we should use ```Zeta.of(context).colors.primary```. You'll need to add a prop for either BuildContext or ZetaColors

File: lib/src/components/progress/progress_circle.dart [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    super.key,
+    super.progress = 0,
+    this.size = ZetaCircleSizes.xl,
+    this.rounded = true,
+  });
+
+  ///Size of [ZetaProgressCircle]
+  final ZetaCircleSizes size;
+
+  /// Border Type for [ZetaWidgetBorder]
```
Comment (thelukewalton) [inline lib/src/components/progress/progress_circle.dart:39]: We have a macro for rounded comment, change this comment to 
``` /// {@macro zeta-component-rounded}```

File: example/widgetbook/pages/components/progress_widgetbook.dart [hunk 5]
```diff
@@ -9,13 +9,38 @@ Widget progressBarUseCase(BuildContext context) => WidgetbookTestWidget(
...
             rounded: context.knobs.boolean(label: 'Rounded'),
             label: context.knobs.stringOrNull(label: 'Label'),
           ),
         );
       }),
     );
+
+Widget progressCircleUseCase(BuildContext context) =>
+    WidgetbookTestWidget(widget: LayoutBuilder(builder: (context, constraints) {
+      return SizedBox(
```
Comment (thelukewalton) [inline example/widgetbook/pages/components/progress_widgetbook.dart:29]: This sized box and LayoutBuilder arent needed, and should be removed

**Review by thelukewalton — APPROVED** [review]

**Pull request merged**